oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/OAuth2TokenAttributes.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.core;
+
+import org.springframework.util.Assert;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author Clement Ng
+ *
+ */
+public class OAuth2TokenAttributes {
+	private Map<String, Object> attributes;
+
+	public OAuth2TokenAttributes() {
```

Comment:
I added the default constructor because `constructorWhenAttributesAreNullOrEmptyThenThrowsException()` tests that an empty attributes map throws an illegal argument exception, but the setup for all the tests originally set `attributes` to an empty map (which is OK because it doesn't go through the constructor for `OAuth2IntrospectionAuthenticationToken`). One potential alternative is to have the ``OAuth2IntrospectionAuthenticationToken.attributes` method do the checking that the map is not empty and the `OAuth2TokenAttributes` are not null. Would this be better?